### PR TITLE
update index.php to eliminate deprecation warning for PHP 8.2

### LINKED ISSUE #50: Deprecation
On auth page displays
```
Deprecated: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /storage/emulated/0/Download/selfauth/index.php on line 371
```
and after entering password
```
Deprecated: base64_encode(): Passing null to parameter #1 ($string) of type string is deprecated in /storage/emulated/0/Download/selfauth/index.php on line 135

Warning: Cannot modify header information - headers already sent by (output started at /storage/emulated/0/Download/selfauth/index.php:135) in /storage/emulated/0/Download/selfauth/index.php on line 334
```
PHP 8.2.8

### PR DESCRIPTION
**Pull request by sugardave**
Should fix https://github.com/Inklings-io/selfauth/issues/50#issue-1833900779

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -132,7 +132,7 @@ function get_q_value($mime, $accept)
 
 function base64_url_encode($string)
 {
-    $string = base64_encode($string);
+    $string = base64_encode($string ?? '');
```
Comment (Zegnat) [inline index.php:135]:
```suggestion
    $string = base64_encode(isset($string) ? $string : "");
```

**Review by Zegnat — CHANGES_REQUESTED** [review]
Note that selfauth is compatible with PHP 5.6. I am not sure we are currently thinking of dropping that backwards compatibility because of the earlier audience for this project.

The null coalescing operator used in this PR [was first introduced in PHP 7.0](https://www.php.net/manual/en/migration70.new-features.php). Could we use a PHP 5.6 compatible solution instead?

File: index.php (hunk 1 above)
Comment (sugardave) [reply index.php:135]: I never saw this comment, oops!  I have moved off my indieweb/selfauth project for a bit, but should hopefully be back to it soon and I can revisit this.

**Comment by dissolve** [comment]
since it has been so long and the needed changes were not updated, I am closing this and have created a separate pull request with the needed change

**Pull request closed without merging**